packages/skeleton-svelte/src/lib/components/Segment/SegmentItem.svelte

```diff
@@ -12,54 +12,33 @@
 		active = 'preset-filled',
 		hover = 'hover:preset-tonal',
 		classes = '',
-		// Input
-		radioBase = 'hidden absolute pointer-events-none',
 		// Label
 		labelBase = 'pointer-events-none',
 		labelClasses = '',
 		// Events
-		onchange = () => {},
+		onclick = () => {},
 		// Snippets
 		children
 	}: SegmentItemProps = $props();
 
-	// Ref
-	let elemCheckbox: HTMLInputElement;
-
-	// Reactive
-	const checked = $derived(group === value);
-	const rxActive = $derived(checked ? active : hover);
+	// Context
+	const ctx = getSegmentContext();
 
 	function onClickHandler() {
-		elemCheckbox.checked = true;
-		group = value;
+		ctx.onSelectionHandler(value);
+		if (onclick) onclick(value);
 	}
+
+	// Reactive
+	const selected = $derived(value === ctx.value);
+	const rxActive = $derived(selected ? active : hover);
 </script>
 
-<button
-	role="radio"
-	aria-checked={checked}
-	onclick={onClickHandler}
-	type="button"
-	class="{base} {rxActive} {classes}"
-	{title}
-	{disabled}
-	data-testid="segment-item"
->
-	<!-- Radio Input (hidden) -->
-	<input
-		class={radioBase}
-		type="radio"
-		bind:this={elemCheckbox}
-		bind:group
-		{name}
-		{value}
-		{checked}
-		onchange={() => onchange(group)}
-		tabindex="-1"
-	/>
+<button onclick={onClickHandler} type="button" class="{base} {rxActive} {classes}" {title} {disabled} data-testid="segment-item">
```

Comment:
We only add ARIA when needed. They're no longer needed because this is no longer taking the role of a radio input functionally. The hidden input operates in a different manner so this would likely not be accurate.

It's worse to put bad ARIA than it is to put no ARIA, so this has been removed intentionally.

Reply:
This pattern where one is selected from multiple is the radio group pattern and we should enforce radio group pattern aria attributes.